### PR TITLE
Add bulkData() method

### PR DESCRIPTION
**Pull request by codetheweb**
Adds a method for the [bulk data](https://scryfall.com/docs/api/bulk-data) endpoint.

Thanks for your work on this, the interface is very nicely designed. 👍 

(Note: my editor automatically trims extra spaces on lines when saving files, so `DOCUMENTATION.md` has quite a few line changes where a single space was removed.)

### CODE REVIEW

File: src/api/Misc.ts [hunk 1]
```diff
@@ -10,8 +10,25 @@ export interface HomepageLink {
...
+	type: string;
+  name: string;
+  description: string;
+  permalink_uri: string;
+  created_at: string;
+  updated_at: string;
+  compressed_size: number;
+  content_type: string;
+  content_encoding: string;
+}
```
Comment (Yuudaari) [inline src/api/Misc.ts]: Fix indent & add object discriminator
```suggestion
export interface BulkData {
	object: "bulk_data";

	id: string;
	type: string;
	name: string;
	description: string;
	permalink_uri: string;
	created_at: string;
	updated_at: string;
	compressed_size: number;
	content_type: string;
	content_encoding: string;
}
```

File: src/api/Misc.ts [hunk 2]
```diff
@@ -10,8 +10,25 @@ export interface HomepageLink {
...
+
 export default new class Misc extends MagicQuerier {
 	public async homepageLinks () {
 		return (await this.query<List<HomepageLink>>("homepage-links")).data;
 	}
+
+  public async bulkData () {
```
Comment (Yuudaari) [inline src/api/Misc.ts]: Fix indent
```suggestion
	public async bulkData () {
```

File: src/tests/Main.ts [hunk 3]
```diff
@@ -440,11 +440,21 @@ describe("Scry", function () {
...
+      expect(result.length).eq(5);
+      expect(Scry.error()).eq(undefined);
+    });
+
+    it("homepage links", async () => {
+      const result = await Scry.Misc.homepageLinks();
+      expect(result).satisfies(Array.isArray);
+      expect(Scry.error()).eq(undefined);
+    });
+  })
```
Comment (Yuudaari) [inline src/tests/Main.ts]: Fix indent, add missing semicolon, replace bulk data length check with `gte` for a bit of future proofing
```suggestion
	describe("Misc", () => {
		it("bulk data", async () => {
			const result = await Scry.Misc.bulkData();

			expect(result).satisfies(Array.isArray);
			expect(result.length).gte(5);
			expect(Scry.error()).eq(undefined);
		});

		it("homepage links", async () => {
			const result = await Scry.Misc.homepageLinks();
			expect(result).satisfies(Array.isArray);
			expect(Scry.error()).eq(undefined);
		});
	});
```

File: DOCUMENTATION.md [hunk 4]
```diff
@@ -432,28 +433,34 @@ Scry.Catalog.watermarks().then(result => console.log(result.length)); // 50
 Scry.homepageLinks().then(result => console.log(result.length)); // 4
 ```
 
+### `bulkData (): Promise<string[]>;` [🡅](#table-of-contents)
```
Comment (Yuudaari) [inline DOCUMENTATION.md]: Incorrect return
```suggestion
### `bulkData (): Promise<BulkData[]>;` [🡅](#table-of-contents)
```

File: DOCUMENTATION.md [hunk 5]
```diff
@@ -43,10 +43,11 @@
   - [`Catalog.watermarks (): Promise<string[]>;` ](#catalogwatermarks--promisestring-)
 - [Misc](#misc-)
   - [`homepageLinks (): Promise<string[]>;`](#homepagelinks--promisestring-)
+  - [`bulkData (): Promise<string[]>;`](#bulkdata--promisestring-)
```
Comment (Yuudaari) [inline DOCUMENTATION.md]: Incorrect return
```suggestion
  - [`bulkData (): Promise<BulkData[]>;`](#bulkdata--promisebulkdata-)
```

File: src/tests/Main.ts (hunk 3 above)
Comment (codetheweb) [reply src/tests/Main.ts]: Good point.

**Comment by codetheweb** [comment]
Thank you, should have ran `tslint` first.

Looks good to me.

**Pull request merged**